sql/mysql/migrate.go

```diff
@@ -424,6 +416,30 @@ func (*state) attr(b *sqlx.Builder, attrs ...schema.Attr) {
 	}
 }
 
+// columnDefault writes the default value of column to the builder.
+func (s *state) columnDefault(b *sqlx.Builder, c *schema.Column) {
+	switch x := c.Default.(type) {
+	case *schema.Literal:
+		v := x.V
+		if !hasNumericDefault(c.Type.Type) && !isHex(v) {
+			v = quote(v)
+		}
+		b.P("DEFAULT", v)
+	case *schema.RawExpr:
+		v := x.X
+		// For backwards compatibility, quote raw expressions that are not wrapped
+		// with parens for non-numeric column types (i.e. literals).
+		switch t := c.Type.Type; {
+		case isHex(v), hasNumericDefault(t), strings.HasPrefix(v, "(") && strings.HasSuffix(v, ")"):
```

Comment:
is this just for future use?

Reply:
No, it is to avoid adding their negation in the if below. It will make it too long.